Add AlmaLinux 10 support

aokiam opened the pull request:
unit tests pass, inspec tests fail

Commit:
added EL10

Commit:
removed Berksfile.lock and ~/berks/cookbooks/osl-repo*, and changed gpgkey url back to pass unit tests

ramereth commented on resources/install.rb:
```diff
@@ -4,7 +4,7 @@
 
 property :yum_baseurl, String, default: 'https://jfut.integ.jp/linux/ganeti/$releasever/$basearch'
-property :yum_gpgkey, String, default: 'https://raw.githubusercontent.com/jfut/ganeti-rpm/refs/heads/main/rpmbuild/integ-ganeti-release-10/SOURCES/RPM-GPG-KEY-integ-ganeti-10'
+property :yum_gpgkey, String, default: 'https://jfut.integ.jp/linux/ganeti/RPM-GPG-KEY-integ-ganeti'
```
Can you please create a helper method which has EL10 using the other key but otherwise use the current key?

I created an [upstream issue](https://github.com/jfut/ganeti-rpm/issues/73) get this added on the mirror so we'll need to change this again before we merge it.

ramereth commented on Berksfile:
```diff
@@ -1,4 +1,5 @@
 source 'https://supermarket.chef.io'
+source 'https://supermarket.osuosl.org'
```
Please remove the github references below since you're adding our supermarket.

Commit:
added helper for EL10 gpgkey support

Commit:
removed github references from Berksfile

ramereth commented on resources/install.rb:
```diff
@@ -4,7 +4,13 @@
-property :yum_gpgkey, String, default: 'https://jfut.integ.jp/linux/ganeti/RPM-GPG-KEY-integ-ganeti'
+property :yum_gpgkey, String, default: lazy {
+                                         if node['platform_version'].to_i == 10
+                                           'https://raw.githubusercontent.com/jfut/ganeti-rpm/refs/heads/main/rpmbuild/integ-ganeti-release-10/SOURCES/RPM-GPG-KEY-integ-ganeti-10'
```
They finally published this, please update it to this:
```suggestion
                                           'https://jfut.integ.jp/linux/ganeti/RPM-GPG-KEY-integ-ganeti-10'
```

ramereth committed:
Add SELinux policy module for ganeti execmem permission

ganeti-wconfd and ganeti-luxid (Haskell binaries) require execmem
permission to run under SELinux enforcing mode. Without this, they
repeatedly fail to start with AVC denials for the execmem permission
on the init_t domain.

Add a custom SELinux policy module (ganeti.te) that grants init_t the
execmem process permission, and install it using the native
selinux_module resource.

Changes:
- files/ganeti.te: SELinux Type Enforcement module allowing
  init_t self:process execmem
- resources/install.rb: Add selinux_module resource to compile and
  install the policy module during ganeti_install
- spec/unit/resources/install_spec.rb: Add unit test for the new
  selinux_module resource
- test/integration/default/controls/default_spec.rb: Add integration
  test verifying the module is loaded via semodule -l

Also include kitchen.yml and kitchen.openstack.yml configuration
updates for AlmaLinux 10 support and Vagrant/cinc provisioner setup.

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Update AlmaLinux 10 GPG key URL to official integ.jp location

The integ-ganeti GPG key for EL10 is now published at the canonical
jfut.integ.jp URL, replacing the temporary GitHub raw content link.

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth reviewed (approved):
Minor bump

aokiam commented:
!bump minor ~

osuosl-manatee commented:
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v6.3.0.  Have a nice day!
